core/compiler.common/src/org/jetbrains/kotlin/types/SmartcastStability.kt

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2010-2021 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.types
+
```

Comment:
OK, added a doc explaning it. I don't think we should cover all the enums present in FE1.0. Nor would it be worth the effort to unify the FE1.0 implementation with FIR. I am open to move the package to other places. I put it here because I need it to be visible from both the FIR tree and the FIR tree gen.